### ISSUE TITLE
Nano: overall dataloader format provided by nano

### ISSUE THREAD

**Issue by zhentaocc** [intel-analytics/BigDL#4209]
## Description
INC has specific data format for the calibration as below:
- (inputs, labels), both inputs and labels should be either a Tensor or a tuple of tensors
- inputs, a tuple of tensors
In #4159 , I resolved this issue typically when users try to use unacceptable dataloaders for INC quantization by raising error and tell users to make proper modifications to their dataloaders.  
While considering the whole pipeline, if users have an INC acceptable dataloader, but before quantizing the model, they use `trainer.compile(..)` to generate a pl model and try to train it by `trainer.fit(...)`, then we will get another error.

## Current Behavior
With #4159 resolved, multiple inputs networks works fine with quantization:
```python
class MultipleInputs(nn.Module):
    def __init__(self):
        super().__init__()
        self.linear1 = nn.Linear(3, 1)
        self.linear2 = nn.Linear(4, 1)

    def forward(self, x1, x2):
        y1 = self.linear1(x1)
        y2 = self.linear2(x2)
        return y1.squeeze()

dataset = TensorDataset((x1, x2), y1)
dataloader = DataLoader(dataset, batch_size=5)

trainer = Trainer()
pl_model = trainer.compile(model, ...)
trainer.quantize(model, dataloader)
```
while before `trainer.quantize(...)`, we usually try to train the model by:
```python
trainer = Trainer()
pl_model = trainer.compile(model, ...)
trainer.fit(model, dataloader)
```
This raises error, because we don't support `((x1, x2), y1)` with our compiled pl model. 
## Expected Behavior
We can successfully run script below:
```python
class MultipleInputs(nn.Module):
    def __init__(self):
        super().__init__()
        self.linear1 = nn.Linear(3, 1)
        self.linear2 = nn.Linear(4, 1)

    def forward(self, x1, x2):
        y1 = self.linear1(x1)
        y2 = self.linear2(x2)
        return y1.squeeze()

dataset = TensorDataset((x1, x2), y1)
dataloader = DataLoader(dataset, batch_size=5)

trainer = Trainer()
pl_model = trainer.compile(model, ...)
trainer.fit(pl_model, dataloader)
trainer.quantize(model, dataloader)
```
## Possible Solution

- For users who don't use our `compile`, instead they have own pl models:
We can only raise error and give instructions for them to modify their dataloader, or sometimes modify model `forward`.
 This should be already done in #4159 .
- For users who use our `compile`:
Perhaps add extra step forcing users to convert their dataloader into a DataModule.
```python
trainer = Trainer()
pl_model = trainer.compile(model, ...)
datamodule = trainer.compile_dataloader(train_loaders, val_loaders, ..., **kwargs)
trainer.fit(pl_model, datamodule)
```
During above compilation, we can have help wrap data like` (x1,x2,y1,y2) `into `(x1,x2), (y1,y2)` by index, and pre-check if yielding data match the model.

**Comment by jason-dai** [comment]
I don't think we need to fix it at this moment; we can simply document the requirement for `compile` and `quantize` with example codes for now.
